### PR TITLE
fix: fix the memory collection within the container

### PR DESCRIPTION
**Pull request by skyoct**
fix: #921

### CODE REVIEW

**Comment by dmthuc** [comment]
Hi @zeromem , please help to take a look at this PR

File: src/Common/ResourceMonitor.cpp [hunk 1]
```diff
@@ -232,12 +260,18 @@ MemoryMonitor::~MemoryMonitor()
 std::optional<MemoryMonitor::Data> MemoryMonitor::getContainerData()
 {
     Data data{};
-    auto mem_usage_val = getNumberFromFile<UInt64>(mem_usage_fs);
+    auto mem_usage_in_bytes_val = getNumberFromFile<UInt64>(mem_usage_fs);
     auto mem_limit_val = getNumberFromFile<UInt64>(mem_limit_fs);
-    if (mem_usage_val && mem_limit_val)
+
+    auto mem_total_inactive_file_val = getNumberFromFileByName<UInt64>(mem_stat_fs, "total_inactive_file");
```
Comment (luffyhwl) [inline src/Common/ResourceMonitor.cpp:266]: Need to check if this value is valid, the return type of getNumberFromFileByName function is std::optinal<T>.

Comment (skyoct) [reply src/Common/ResourceMonitor.cpp:265]: Hi @luffyhwl, Thank you for helping with the review, I have added validation.

File: src/Common/ResourceMonitor.cpp [hunk 2]
```diff
@@ -232,12 +260,17 @@ MemoryMonitor::~MemoryMonitor()
...
-    auto mem_usage_val = getNumberFromFile<UInt64>(mem_usage_fs);
+    auto mem_usage_in_bytes_val = getNumberFromFile<UInt64>(mem_usage_fs);
     auto mem_limit_val = getNumberFromFile<UInt64>(mem_limit_fs);
-    if (mem_usage_val && mem_limit_val)
+    auto mem_total_inactive_file_val = getNumberFromFileByName<UInt64>(mem_stat_fs, "total_inactive_file");
+
+    if (mem_usage_in_bytes_val && mem_limit_val && mem_total_inactive_file_val)
     {
+        // referring to the memory usage method in k8s, use memory.usage_in_bytes - total_inactive_file.
+        auto mem_working_usage_val = *mem_usage_in_bytes_val - *mem_total_inactive_file_val;
```
Comment (luffyhwl) [inline src/Common/ResourceMonitor.cpp:270]: According to this blog(https://cloud.tencent.com/developer/article/1637682), perhaps we still need to subtract memory.stat.**total_active_file**?

**Comment by luffyhwl** [comment]
LGTM

**Pull request merged**